src/realm/sync/noinst/sync_schema_migration.cpp

```diff
@@ -106,4 +116,45 @@ void track_sync_schema_migration(Transaction& wt, uint64_t previous_schema_versi
     }
 }
 
+void perform_schema_migration(DBRef db, sync::SubscriptionStore& store)
+{
+    // Everything is performed in one single write transaction.
+    auto tr = db->start_write();
+
+    // Disable sync replication.
+    auto& repl = dynamic_cast<sync::ClientReplication&>(*db->get_replication());
+    sync::TempShortCircuitReplication sync_history_guard(repl);
+
+    // Delete all public tables (and their columns).
+    const bool ignore_backlinks = true;
+    for (const auto& tk : tr->get_table_keys()) {
+        // Do not remove the metadata tables
```

Comment:
Do we actually want to leave the private tables? There shouldn't be anything that needs to be preserved, and deleting and recreating them is closer in behavior to what deleting the file did.

Reply:
Not really. I initially deleted those as well, but there was an issue with it. Just got an idea how I may be able to overcome that.